ext/nnstreamer/tensor_decoder/tensordec-tensor_region.c

```diff
@@ -0,0 +1,776 @@
+  GstTensorMetaInfo meta;
+  GstMapInfo out_info;
+  GstMemory *out_mem;
+  GArray *results = NULL;
+  const guint num_tensors = config->info.num_tensors;
+  gboolean need_output_alloc = gst_buffer_get_size (outbuf) == 0;
+  const size_t size = (size_t) 4 * trData->num * sizeof(uint32_t); /**4 field per block */
+
+  g_assert (outbuf);
+  /**Ensure we have outbuf properly allocated */
+  if (need_output_alloc) {
+    out_mem = gst_allocator_alloc (NULL, size, NULL);
+  } else {
+    if (gst_buffer_get_size (outbuf) < size) {
+      gst_buffer_set_size (outbuf, size);
+    }
+    out_mem = gst_buffer_get_all_memory (outbuf);
+  }
+  if (!gst_memory_map (out_mem, &out_info, GST_MAP_WRITE)) {
+    ml_loge ("Cannot map output memory / tensordec-tensor_region.\n");
+    goto error_free;
+  }
+
+  /** reset the buffer with 0 */
+  memset (out_info.data, 0, size);
+  if(trData->mode == MOBILENET_SSD_BOUNDING_BOX){
+    const GstTensorMemory *boxes, *detections = NULL;
+    properties_MOBILENET_SSD *data = &trData->mobilenet_ssd;
+
+    g_assert (num_tensors >= MOBILENET_SSD_MAX_TENSORS);
+    results = g_array_sized_new (FALSE, TRUE, sizeof (detected_object), 100);
+
+    boxes = &input[0];
+    if (num_tensors >= MOBILENET_SSD_MAX_TENSORS) /**lgtm[cpp/constant-comparison] */
+      detections = &input[1];
+    
+    switch (config->info.info[0].type) {
+      _get_objects_mobilenet_ssd_ (uint8_t, _NNS_UINT8);
+      _get_objects_mobilenet_ssd_ (int8_t, _NNS_INT8);
+      _get_objects_mobilenet_ssd_ (uint16_t, _NNS_UINT16);
+      _get_objects_mobilenet_ssd_ (int16_t, _NNS_INT16);
+      _get_objects_mobilenet_ssd_ (uint32_t, _NNS_UINT32);
+      _get_objects_mobilenet_ssd_ (int32_t, _NNS_INT32);
+      _get_objects_mobilenet_ssd_ (uint64_t, _NNS_UINT64);
+      _get_objects_mobilenet_ssd_ (int64_t, _NNS_INT64);
+      _get_objects_mobilenet_ssd_ (float, _NNS_FLOAT32);
+      _get_objects_mobilenet_ssd_ (double, _NNS_FLOAT64);
+      default:
+        g_assert (0);
+    }
+
+    nms (results, data->params[MOBILENET_SSD_PARAMS_IOU_THRESHOLD_IDX]);
+  } else {
+    GST_ERROR ("Failed to get output buffer, unknown mode %d.", trData->mode);
+    goto error_unmap;
+  }
+  gst_tensor_top_detectedObjects_cropInfo (&out_info, trData, results);
+
+  g_array_free (results, TRUE);
+
+  gst_memory_unmap (out_mem, &out_info);
+
+  /** converting to Flexible tensor since  
+   * info pad of tensor_crop has capability for flexible tensor stream 
+  */
+  init_meta(&meta, trData);
+  out_mem = gst_tensor_meta_info_append_header(&meta, out_mem);
+
+  if (need_output_alloc)
+    gst_buffer_append_memory (outbuf, out_mem);
+  else
+    gst_memory_unref (out_mem);
+
+  return GST_FLOW_OK;
+error_unmap:
+  gst_memory_unmap (out_mem, &out_info);
+error_free:
+  gst_memory_unref (out_mem);
+
+  return GST_FLOW_ERROR;
+}
+
+/**
+ * @brief set the max_detection
+ */
+static int
+_set_max_detection (tensor_region *data, const guint max_detection, const unsigned int limit)

```

Comment:
You don't need to constify scalar arguments unlike pointers that are recommended to be constified.